index.d.ts

```diff
@@ -1,8 +1,8 @@
 declare namespace hubdown {
   interface Options {
-    readonly runBefore: Array<any>;
-    readonly frontmatter: boolean;
-    readonly cache: any;
+    readonly runBefore?: Array<any>;
```

Comment:
If you're interested, [`unified`](https://github.com/unifiedjs/unified) (and by extension [`remark`](https://github.com/remarkjs/remark)) has a [`Plugin` type](https://github.com/unifiedjs/unified/blob/12e0adc5b5240ebf1ec378a22a7b6e2f64a266d7/types/index.d.ts#L205-L217) that could be used here.

Reply:
In this case https://github.com/unifiedjs/unified/blob/12e0adc5b5240ebf1ec378a22a7b6e2f64a266d7/types/index.d.ts#L273-L278 may be more accurate

```ts
import {PluggableList} = require('unified');
import {RemarkOptions} = require('remark');

// ...
   readonly runBefore?: PluggableList<Partial<RemarkOptions>>;
```